simdjson-flags.cmake

```diff
@@ -35,6 +35,7 @@ set(THREADS_PREFER_PTHREAD_FLAG ON)
 # Flags used by exes and by the simdjson library (project-wide flags)
 #
 add_library(simdjson-flags INTERFACE)
+add_library(simdjson-public-flags INTERFACE)
```

Comment:
Given the naming scheme you used in simdjson-internal, should we make these simdjson-internal-flags and simdjson-flags, respectively? I like the idea of reserving the "good names" for more public-facing things.

Reply:
Yes, that would work better. It was just me being lazy. Instead of renaming all other references to simdjson-flags I thought it would be easier to just add a new one but I'll change accordingly